addons/web/static/src/js/fields/abstract_field.js

```diff
@@ -299,6 +299,32 @@ var AbstractField = Widget.extend({
         return this._render() || $.when();
     },
 
+    /**
+     * add the invalid class on a field
+     */
+    setInvalidClass: function () {
+        this.$el.toggleClass('o_field_invalid',true);
+    },
+
+    /**
+     * Remove the invalid class on a field
+     */
+    removeInvalidClass: function () {
+        this.$el.toggleClass('o_field_invalid',false);
+    },
+
+    /**
+     * Update the modifiers with the newest value. 
+     * Now this.attrs.modifiersValue can be used consistantly even with conditional modifiers
+     * 
+     * @param {Object} modifiers  the updated modifiers
+     * @override
+     */
+    updateModifiersValue: function(modifiers) {
```

Comment:
if you make the field widget aware of its modifier value, maybe the isValid method should use it and the rest of the system could be changed as well?